test/System/Console/GetOpt/GenericsSpec.hs

```diff
@@ -29,58 +32,76 @@ spec :: Spec
 spec = do
   describe "withArguments" $ do
     it "parses command line arguments" $ do
-      withArgs (words "--bar 4 --baz foo") $ withArguments $ \ options ->
-        options `shouldBe` Foo (Just 4) "foo" False
+      withArgs (words "--bar 4 --baz foo") $
+        getArguments `shouldReturn` Foo (Just 4) "foo" False
 
     it "allows optional arguments" $ do
-      withArgs (words "--baz foo") $ withArguments $ \ options ->
-        options `shouldBe` Foo Nothing "foo" False
+      withArgs (words "--baz foo") $
+        getArguments `shouldReturn` Foo Nothing "foo" False
 
     it "allows boolean flags" $ do
-      withArgs (words "--bool --baz foo") $ withArguments $ \ options ->
-        options `shouldBe` Foo Nothing "foo" True
+      withArgs (words "--bool --baz foo") $
+        getArguments `shouldReturn` Foo Nothing "foo" True
 
     context "invalid arguments" $ do
       it "doesn't execute the action" $ do
-        let main = withArgs (words "--invalid") $
-              withArguments $ \ (_ :: Foo) ->
-                throwIO (ErrorCall "action")
+        let main = withArgs (words "--invalid") $ do
+              _ :: Foo <- getArguments
+              throwIO (ErrorCall "action")
```

Comment:
It tests that `throwIO (ErrorCall "action")` will not be performed. Instead `ExitFailure 1` is thrown.